data-updates/notebooks/app_ees_downloads.r

```diff
@@ -0,0 +1,306 @@
+  rename("publication" = title)
+
+dates <- create_dates(max(downloads$date))
+
+test_that("There are no missing dates since we started", {
+  expect_equal(
+    setdiff(downloads$date, seq(as.Date(dates$all_time_date), max(dates$latest_date), by = "day")) |>
+      length(),
+    0
+  )
+})
+
+# COMMAND ----------
+
+# MAGIC %sql
+# MAGIC select * from
+# MAGIC catalog_40_copper_statistics_services.analytics_app.ees_downloads
+# MAGIC where publication is null
+
+# COMMAND ----------
+
+# selecting just the columns we're interested in storing
+# TO DO: decide if we only want subsets of page_types in here (e.g make it just about publications or remove defunct pages like data catalogue)
+
+downloads <- downloads %>%
+select(date, pagePath, page_type, download_type, publication, eventLabel, eventCount)
+
+# COMMAND ----------
+
+# DBTITLE 1,Write out app data
+updated_spark_df <- copy_to(sc, downloads, overwrite = TRUE)
+
+# Write to temp table while we confirm we're good to overwrite data
+spark_write_table(updated_spark_df, paste0(write_table_name, "_temp"), mode = "overwrite")
+
+temp_table_data <- sparklyr::sdf_sql(sc, paste0("SELECT * FROM ", write_table_name, "_temp")) %>% collect()
+previous_data <- tryCatch(
+  {
+    sparklyr::sdf_sql(sc, paste0("SELECT * FROM ", write_table_name)) %>% collect()
+  },
+  error = function(e) {
+    NULL
+  }
+)
+
+test_that("Temp table data matches updated data", {
+  expect_equal(temp_table_data, downloads)
+})
+
+# Replace the old table with the new one
+dbExecute(sc, paste0("DROP TABLE IF EXISTS ", write_table_name))
+dbExecute(sc, paste0("ALTER TABLE ", write_table_name, "_temp RENAME TO ", write_table_name))
+
+print_changes_summary(temp_table_data, previous_data)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC We're left with the following table 
+# MAGIC
+# MAGIC - **date**: The date the event occured on (earliest date = 21/04/2021)
+# MAGIC - **pagePath**: The pagePath the event occured on
+# MAGIC - **page_type**: Type of service page (Release page, Data catalogue, permalink etc)
+# MAGIC - **download_type**: Type of download (csv, ods, all files etc)
+# MAGIC - **publication**: The publication title (relevant for pages that have an associated publication in their pagePath only)
+# MAGIC TO DO: for some events we can take publicaiton details from the eventLabel and for permalinks we may be able to get publication from a scrape or the EES database - not doing anythign with these atm! 
+# MAGIC - **eventLabel**: The info we have for what file was downloaded (this often includes the relevant publication too though is truncated unhelpfully)
+# MAGIC - **eventCount**: The number of downloads of that type on given day
+# MAGIC
+
+# COMMAND ----------
+
+# MAGIC %sql
+# MAGIC select page_type, download_type, sum(eventCount)
+# MAGIC from catalog_40_copper_statistics_services.analytics_app.ees_downloads
+# MAGIC group by page_type, download_type
+# MAGIC order by page_type, download_type
```

Comment:
Is this an intentional comment / SQL query or one you used to manually test as you went?